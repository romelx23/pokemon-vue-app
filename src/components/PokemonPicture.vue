<script setup>
import { computed } from "@vue/runtime-core";
const { pokemonid } = defineProps({
    pokemonid: {
        type: Number,
        default: 4,
        required: true,
    },
    showPokemon: {
        type: Boolean,
        default: false,
        required: true,
    },
});
const image = computed(() => {
    return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${pokemonid}.svg`;
});
</script>

<template>
    <div class="pokemon-image">
        <img class="filter pokemon" :src="image" alt="pokemon" title="pokemon">
        <img v-if="showPokemon" class="fade-in pokemon" :src="image" alt="pokemon" title="pokemon">
    </div>
</template>

<style scoped>
.pokemon-image {
    width: 100%;
    height: 250px;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
}

img {
    position: absolute;
}

.pokemon {
    width: auto;
    height: 100%;
}

.filter {
    filter: brightness(0);
    pointer-events: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}

.fade-in {
    animation: fade-in 1s;
}

@keyframes fade-in {
    from {
        opacity: 0;
    }

    to {
        opacity: 1;
    }
}
</style>