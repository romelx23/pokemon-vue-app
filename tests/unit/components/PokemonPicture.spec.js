import { describe, expect } from "vitest";
import PokemonPicture from "@/../../src/components/PokemonPicture.vue";
import { mount, shallowMount } from "@vue/test-utils";

describe("PokemonPicture", () => {
  test("debe de hacer match con el SnapShot", () => {
    const wrapper = mount(PokemonPicture, {
      props: {
        pokemonid: 1,
        showPokemon: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("debe de mostrar la imagen oculta y el pokemon 100", () => {
    const wrapper = mount(PokemonPicture, {
      props: {
        pokemonid: 100,
        showPokemon: false,
      },
    });
    const [img1, img2] = wrapper.findAll("img");
    expect(img1.exists()).toBeTruthy();
    expect(img2).toBe(undefined);
    console.log(img1.classes());
    // console.log(img2.classes());
    expect(img1.classes("filter")).toBeTruthy();
    // https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg
    expect(img1.attributes("src")).toBe(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
    );
  });

  test("debe de mostrar la imagen del pokemon si showPokemon:true", () => {
    const wrapper = mount(PokemonPicture, {
      props: {
        pokemonid: 100,
        showPokemon: true,
      },
    });
    const img1 = wrapper.find("img");
    expect(img1.exists()).toBeTruthy();
    // console.log(img2.classes());
    expect(img1.classes("pokemon")).toBe(true);
    expect(img1.classes("fade-in")).toBe(false);
  });
});
