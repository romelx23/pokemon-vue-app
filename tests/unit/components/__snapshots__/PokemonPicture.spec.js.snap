// Vitest Snapshot v1

exports[`PokemonPicture > debe de hacer match con el SnapShot 1`] = `"<div class=\\"pokemon-image\\" data-v-c76ad58e=\\"\\"><img class=\\"filter pokemon\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg\\" alt=\\"pokemon\\" title=\\"pokemon\\" data-v-c76ad58e=\\"\\"><img class=\\"fade-in pokemon\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg\\" alt=\\"pokemon\\" title=\\"pokemon\\" data-v-c76ad58e=\\"\\"></div>"`;
